src/main/java/decodex/modules/rot/RotEncoder.java

```diff
@@ -0,0 +1,93 @@
+package decodex.modules.rot;
+
+import decodex.data.Data;
+import decodex.modules.Module;
+
+/**
+ * The RotEncoder class handles the rotation of the alphabets in the input string.
+ */
+public class RotEncoder extends Module {
+
+    public static final String MODULE_NAME = "rotencode";
+    public static final String MODULE_DESCRIPTION = "Rotates the alphabets in the input string by a specified amount";
+
+    private static final char LOWERCASE_A = 'a';
+    private static final char LOWERCASE_Z = 'z';
+    private static final char UPPERCASE_A = 'A';
+    private static final char UPPERCASE_Z = 'Z';
+    private static final int ALPHABETS_COUNT = 26;
+
+    private final int rotateOffset;
+
+    public RotEncoder(int rotateOffset) {
+        super(MODULE_NAME, MODULE_DESCRIPTION);
+        this.rotateOffset = rotateOffset;
+    }
+
+    /**
+     * Applies the rotation to the given Data object.
+     *
+     * @param data Data object to encode.
+     * @return new Data object representing the result.

```

Comment:
The description of the return data is not descriptive of the modification done. Consider `A Data object with its alphabetical characters rotated.`.

Reply:
I think a suggestion would be `The Data object containing the rotated result`?